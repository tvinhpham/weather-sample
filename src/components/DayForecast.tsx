import React from 'react';
import { FunctionComponent, HTMLAttributes } from 'react';
import Card from 'react-bootstrap/Card';
import { parseISO, format } from 'date-fns';
import LazyLoadImage from './LazyLoadImage';
import { API_URL } from '../utils/constants';

export type ForecastDetails = {
  id: number;
  applicable_date: string;
  max_temp: number;
  min_temp: number;
  weather_state_abbr: string;
};

const getDayOfWeek = (date: string) => {
  const parsedDate = parseISO(date);
  return format(parsedDate, 'EEEE');
};

const DayForecast: FunctionComponent<{ data: ForecastDetails } & HTMLAttributes<HTMLElement>> = ({
  data,
  ...props
}) => {
  const { weather_state_abbr, applicable_date, min_temp, max_temp } = data;
  return (
    <div {...props}>
      <Card>
        <Card.Header>
          <Card.Title className="text-center">{getDayOfWeek(applicable_date)}</Card.Title>
        </Card.Header>
        <LazyLoadImage src={`${API_URL}/static/img/weather/${weather_state_abbr}.svg`} />
        <Card.Body className="d-flex flex-column">
          <span className="text-center"> Min: {min_temp.toFixed(0)}&#8451;</span>
          <span className="text-center">Max: {max_temp.toFixed(0)}&#8451;</span>
        </Card.Body>
      </Card>
    </div>
  );
};

export default DayForecast;
