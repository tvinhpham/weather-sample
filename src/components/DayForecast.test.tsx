import React from 'react';
import renderer from 'react-test-renderer';
import DayForecast, { ForecastDetails } from './DayForecast';

it('renders ForecastDetails', () => {
  const data:ForecastDetails = {
    id: 1,
    applicable_date: '2021-10-14',
    max_temp: 20.001,
    min_temp: 30.6,
    weather_state_abbr: 'hc'
  };
  const forecast = renderer
    .create(<DayForecast data={data}/>)
    .toJSON();
  expect(forecast).toMatchSnapshot();
});
