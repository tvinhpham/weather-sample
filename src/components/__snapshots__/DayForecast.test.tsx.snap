// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ForecastDetails 1`] = `
<div>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      <div
        className="text-center card-title h5"
      >
        Thursday
      </div>
    </div>
    <div
      className="w-100 position-relative undefined"
      style={
        Object {
          "objectFit": "cover",
          "overflow": "hidden",
          "paddingTop": "100%",
        }
      }
    >
      <div
        className="lazyload-wrapper "
      >
        <div
          className="position-absolute w-100 h-100 spinner-grow"
        />
      </div>
    </div>
    <div
      className="d-flex flex-column card-body"
    >
      <span
        className="text-center"
      >
         Min: 
        31
        ℃
      </span>
      <span
        className="text-center"
      >
        Max: 
        20
        ℃
      </span>
    </div>
  </div>
</div>
`;
